client/update.go

```diff
@@ -411,3 +419,8 @@ func makeChannelUpdate(next *channel.State, actor channel.Index) ChannelUpdate {
 		ActorIdx: actor,
 	}
 }
+
+// Error implements error interface for PeerNotRespondingError.
+func (e PeerNotRespondingError) Error() string {
```

Comment:
Could you add constructors for both error types which also accept an error and use the `WithStack` function?  
Then the context error would be passed on instead of being discarded (like now). You can take a look at `channel/errors.go`.

Reply:
I have added `newRequestTimedOutError` function.